src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,256 @@
+Jordan algebras arising this way are said to be special. There are also exceptional Jordan algebras
+which can be shown not to be the symmetrisation of any associative algebra. The 3x3
+matrices of octonions is the canonical example.
+
+Commutative Jordan algebras were introduced by Jordan, von Neumann and Wigner
+([jordanvonneumannwigner1934]) as a mathematical model for the observables of a quantum mechanical
+physical system (for a C*-algebra the self-adjoint part is closed under the symmetrised Jordan
+multiplication). Jordan algebras have subsequently been studied from the points of view of abstract
+algebra and functional analysis. They have connections to Lie algebras and differential geometry.
+
+A more general concept of a (non-commutative) Jordan algebra can also be defined, as a
+(non-commutative, non-associative) algebra `A` where, for each `a` in `A`, the operators of left and
+right multiplication by `a` and `a^2` commute. Such algebras have connected to the Vidav-Palmer
+theorem [cabreragarciarodriguezpalacios2014].
+
+A comprehensive overview of the algebraic theory can be found in [mccrimmon2004].
+
+A real Jordan algebra `A` can be introduced by
+```
+variables {A : Type*} [non_unital_non_assoc_ring A] [module ℝ A] [smul_comm_class ℝ A A]
+  [is_scalar_tower ℝ A A] [comm_jordan A]
+```
+
+## Main results
+
+- `lin_jordan` : Linearisation of the commutative Jordan axiom
+
+## Implementation notes
+
+We shall primarily be interested in linear Jordan algebras (i.e. over rings of characteristic not
+two) leaving quadratic algebras to those better versed in that theory.
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Jordan, von Neumann and Wigner, 1934][jordanvonneumannwigner1934]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+
+-/
+
+set_option old_structure_cmd true
+
+section non_unital_non_assoc_ring
+
+variables {A : Type*} [non_unital_non_assoc_semiring A]
+
+namespace non_unital_non_assoc_ring
+/--
+Left multiplication operator
+-/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/--
+Right multiplication operator
+-/
+@[simps] def R : A→+(add_monoid.End A) := add_monoid_hom.flip (L : A →+ add_monoid.End A)
+
+lemma L_def (a b : A) : L a b = a*b := rfl
+
+lemma R_def (a b : A) : R a b = b*a := rfl
+
+end non_unital_non_assoc_ring
+
+end non_unital_non_assoc_ring
+
+open non_unital_non_assoc_ring
+/--
+A non unital, non-associative ring with a (non-commutative) Jordan multiplication.
+-/
+class jordan_ring(A : Type*) [non_unital_non_assoc_ring A] :=
```

Comment:
Perhaps:
```suggestion
class is_jordan (A : Type*) [has_mul A] :=
```

Reply:
I originally had something like that - but switched to `non_unital_non_assoc_ring` when I re-wrote the axioms in terms of Lie brackets. Now that they're not stated in those terms I could look at switching back.

Reply:
Done, thanks!